File: frontend/src/types/meeting.ts
export interface Meeting {

    source: string;

    source_id: string;

    title: string;

    client_name?: string;

    client_company?: string;

    owner: string;

    start: string;

    location?: string;

    status: string;

    notes?: string;
}

export interface UnifiedMeeting {

    crm?: Meeting;

    calendar?: Meeting;

    match_score: number;

    conflicts: Record<
        string,
        {
            crm: string;
            calendar: string;
        }
    >;
}

File: backend/app/models/meeting.py
from dataclasses import dataclass
from datetime import datetime
from typing import Optional

@dataclass
class Meeting:
    # Required fields (no defaults) first
    source: str
    source_id: str
    title: str
    owner: str
    
    # Optional fields with defaults
    client_name: Optional[str] = None
    client_company: Optional[str] = None
    start: Optional[datetime] = None
    location: Optional[str] = None
    status: str = "pending"          
    notes: Optional[str] = None
